### ISSUE TITLE
Regression: firewall-7.0.0 breaks chains with hypen in its name

### ISSUE THREAD

**Issue by sircubbi** [puppetlabs/puppetlabs-firewall#1156]
## Describe the Bug
The pattern for jump-targets introduced with 7.0.0 does not allow hypens. This breaks rules created by failban.

## Expected Behavior
It should be possible to exclude chains which have a hyphen in its name, as well as rules which jump to these chains from purging. Therefore `Pattern[/^[a-zA-Z0-9_]+$/]` needs to be changed to `Pattern[/^[-a-zA-Z0-9_]+$/]`

## Environment
 - Version 7.0.0
 - Platform EL7

## Additional Context
Manifest includes following purge-disabling rule:
```
  firewallchain {
    'FORWARD:filter:IPv4':
      purge  => true,
      ignore => [
        '-j f2b-innovaphone-office',
      ];

    'f2b-innovaphone-office:filter:IPv4':
      purge => false;
  }
```

That chain is autocreated by fail2ban, which also adds a rule with a jump to this chain in `FORWARD`.

Parsing that rule fails, because the jump-target includes a hypen:
```
Error: /Stage[main]/Profile::Server::Rx_ipva_prov/Firewallchain[FORWARD:filter:IPv4]: Failed to generate additional resources using 'generate': Provider returned data that does not match the Type Schema for `firewall[9051 cfd13b2858519042a62d816accebac39f4d2f0f281886430dcfe051610d61432]`
 Value type mismatch:
    * jump: f2b-innovaphone-office (expects an undef value or a match for Pattern[/^[a-zA-Z0-9_]+$/], got 'f2b-innovaphone-office')
```

**Comment by david22swan** [comment]
Think it would be best just to swap jump to `String[1]`, probably goto as well

**Comment by sircubbi** [comment]
> Think it would be best just to swap jump to `String[1]`, probably goto as well

That should do the trick, yes.

**Comment by sircubbi** [comment]
Thanks, is fixed with 7.0.1.
